src/swap/useSwapQuote.ts

```diff
@@ -54,21 +56,32 @@ function createBaseSwapTransactions(
       ? BigInt(new BigNumber(buyAmount).times(guaranteedPrice).toFixed(0, 0))
       : BigInt(sellAmount)
 
-  // Approve transaction if the sell token is ERC-20
-  // TODO skip this if the allowance is already enough
-  if (allowanceTarget !== zeroAddress && fromToken.address) {
-    const data = encodeFunctionData({
+  // If the sell token is ERC-20, we need to check the allowance and add an
+  // approval transaction if necessary
+  if (allowanceTarget !== zeroAddress && fromToken.address && isAddress(fromToken.address)) {
```

Comment:
nit: Why checking `isAddress(fromToken.address)`? seems a bit too defensive, as if we were not trusting our own data.

Reply:
the viem functions require the address to be `Address` type, so typescript complains about the address being the `string` type (less specific). i guess broadly we've been using `as Address`, but we discussed recently to avoid type casting so i went with the typeguard here. i think it should make no difference at runtime. i'm not sure what is best, going for consistency + type casting or going with the type guard (which IMO is more correct)?

Reply:
i think consistency is top priority and makes it easier to do a project wide refactor if we want in the future...so i've removed the type guards and did `as Address` in the viem function directly.

Reply:
I think we should turn out tokensInfo address field (and others) to be `Address` this would remain compatible with things not using viem and will avoid the need to casting when using viem.

What do you think?

Reply:
yes i agree - we should update the redux type to be Address, i can create a task so that we don't forget